services/cd-service/pkg/service/overview_test.go

```diff
@@ -45,170 +48,14 @@ func (m *mockOverviewService_StreamOverviewServer) Context() context.Context {
-				})},
-			ExpectedWarnings: []*api.Warning{},
-		},
-		{
-			Name:    "app deployed in higher version on upstream should warn",
-			AppName: "foo",
-			Groups: []*api.EnvironmentGroup{
-				makeEnvGroup(dev, []*api.Environment{
-					makeEnv("prod", dev, makeUpstreamEnv("dev"),
-						makeApps(makeApp("foo", 2))),
-				}),
-				makeEnvGroup(dev, []*api.Environment{
-					makeEnv("dev", dev, makeUpstreamLatest(),
-						makeApps(makeApp("foo", 1))),
-				}),
-			},
-			ExpectedWarnings: []*api.Warning{
-				{
-					WarningType: &api.Warning_UnusualDeploymentOrder{
-						UnusualDeploymentOrder: &api.UnusualDeploymentOrder{
-							UpstreamVersion:     1,
-							UpstreamEnvironment: "dev",
-							ThisVersion:         2,
-							ThisEnvironment:     "prod",
-						},
-					},
-				},
-			},
-		},
-		{
-			Name:    "app deployed in same version on upstream should not warn",
-			AppName: "foo",
-			Groups: []*api.EnvironmentGroup{
-				makeEnvGroup(dev, []*api.Environment{
-					makeEnv("prod", dev, makeUpstreamEnv("dev"),
-						makeApps(makeApp("foo", 2))),
-				}),
-				makeEnvGroup(dev, []*api.Environment{
-					makeEnv("dev", dev, makeUpstreamLatest(),
-						makeApps(makeApp("foo", 2))),
-				}),
-			},
-			ExpectedWarnings: []*api.Warning{},
-		},
-		{
-			Name:    "app deployed in no version on upstream should warn",
-			AppName: "foo",
-			Groups: []*api.EnvironmentGroup{
-				makeEnvGroup(dev, []*api.Environment{
-					makeEnv("prod", dev, makeUpstreamEnv("dev"),
-						makeApps(makeApp("foo", 1))),
-				}),
-				makeEnvGroup(dev, []*api.Environment{
-					makeEnv("dev", dev, makeUpstreamLatest(),
-						makeApps()),
-				}),
-			},
-			ExpectedWarnings: []*api.Warning{
-				{
-					WarningType: &api.Warning_UpstreamNotDeployed{
-						UpstreamNotDeployed: &api.UpstreamNotDeployed{
-							UpstreamEnvironment: "dev",
-							ThisVersion:         1,
-							ThisEnvironment:     "prod",
-						},
-					},
-				},
-			},
-		},
-	}
-	for _, tc := range tcs {
-		tc := tc
-		t.Run(tc.Name, func(t *testing.T) {
-			actualWarnings := CalculateWarnings(testutil.MakeTestContext(), tc.AppName, tc.Groups)
-			if len(actualWarnings) != len(tc.ExpectedWarnings) {
-				t.Errorf("Different number of warnings. got: %s\nwant: %s", actualWarnings, tc.ExpectedWarnings)
-			}
-			for i := 0; i < len(actualWarnings); i++ {
-				actualWarning := actualWarnings[i]
-				expectedWarning := tc.ExpectedWarnings[i]
-				if diff := cmp.Diff(actualWarning.String(), expectedWarning.String()); diff != "" {
-					t.Errorf("Different warning at index [%d]:\ngot:  %s\nwant: %s", i, actualWarning, expectedWarning)
-				}
-			}
-		})
-	}
-
-}
-
 func TestOverviewService(t *testing.T) {
 	var dev = "dev"
 	tcs := []struct {
-		Name  string
-		Setup []repository.Transformer
-		Test  func(t *testing.T, svc *OverviewServiceServer)
+		Name         string
+		Setup        []repository.Transformer
+		Test         func(t *testing.T, svc *OverviewServiceServer)
```

Comment:
same comment about test functions: Please make this its own (top-level) test. There should be no need for functions in a test array.